src/contexts/VideoContext.jsx

```diff
@@ -0,0 +1,35 @@
+import { createContext, useContext, useReducer } from "react";
+import { videoReducer } from "../reducers/videoReducer";
+import {
+  addToWatchlist,
+  removeFromWatchlist,
+} from "../router/utils/HelperFunctions";
+
+const VideoContext = createContext();
+
+const VideoProvider = ({ children }) => {
+  const initialData = {
+    watchList: [],
```

Comment:
You can move this object outside of the function component. Else it will be recreated with each render

Reply:
Done, thank you!